junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/EnabledForJreRangeCondition.java

```diff
@@ -31,13 +31,54 @@ class EnabledForJreRangeCondition extends BooleanExecutionCondition<EnabledForJr
 
 	@Override
 	boolean isEnabled(EnabledForJreRange annotation) {
-		JRE min = annotation.min();
-		JRE max = annotation.max();
+		return isCurrentVersionWithinRange("EnabledForJreRange", annotation.min(), annotation.max(),
+			annotation.minVersion(), annotation.maxVersion());
+	}
+
+	static boolean isCurrentVersionWithinRange(String annotationName, JRE minJre, JRE maxJre, int minVersion,
+			int maxVersion) {
+
+		boolean minJreSet = minJre != JRE.UNDEFINED;
+		boolean maxJreSet = maxJre != JRE.UNDEFINED;
+		boolean minVersionSet = minVersion != JRE.UNDEFINED_VERSION;
+		boolean maxVersionSet = maxVersion != JRE.UNDEFINED_VERSION;
+
+		// Users must choose between JRE enum constants and version numbers.
+		Preconditions.condition(!minJreSet || !minVersionSet, () -> String.format(
+			"@%s's minimum value must be configured with either a JRE enum constant or numeric version, but not both",
+			annotationName));
+		Preconditions.condition(!maxJreSet || !maxVersionSet, () -> String.format(
+			"@%s's maximum value must be configured with either a JRE enum constant or numeric version, but not both",
+			annotationName));
+
+		// Users must supply valid values for minVersion and maxVersion.
+		Preconditions.condition(!minVersionSet || (minVersion >= 8),
```

Comment:
I think we should extract `8` into a constant so it's easier to change when we bump the baseline to 17.